ts/features/bonus/bpd/components/paymentMethodActivationToggle/PaymentMethodBpdToggle.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from "react";
+import { useContext, useEffect, useRef } from "react";
+import { Image, StyleSheet } from "react-native";
+import { NavigationContext } from "react-navigation";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import image from "../../../../../../img/wallet/cards-icons/pagobancomat.png";
+import { Body } from "../../../../../components/core/typography/Body";
+import { GlobalState } from "../../../../../store/reducers/types";
+import {
+  bpdPaymentMethodActivation,
+  BpdPaymentMethodActivation,
+  BpdPmActivationStatus,
+  bpdUpdatePaymentMethodActivation,
+  HPan
+} from "../../store/actions/paymentMethods";
+import { bpdPaymentMethodValueSelector } from "../../store/reducers/details/paymentMethods";
+import { BpdToggle } from "./BpdToggle";
+
+// TODO: accept only hpan, read all the other information with a selector from payment methods
+type OwnProps = { hPan: HPan; hasBpdCapability: boolean };
+
+export type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps> &
+  OwnProps;
+
+type GraphicalState = "loading" | "ready" | "update";
+
+export type GraphicalValue = {
+  state: GraphicalState;
+  value: BpdPmActivationStatus | undefined;
+};
+
+const styles = StyleSheet.create({
+  row: {
+    flex: 1,
+    flexDirection: "row",
+    alignItems: "center",
+    height: 48
+  },
+  cardIcon: {
+    width: 40,
+    height: 25,
+    overflow: "hidden",
+    resizeMode: "contain"
+  },
+  leftSection: { flexDirection: "row", flex: 1 }
+});
+
+const retryTimeout = 5000 as Millisecond;
+/**
+ * This custom hook handles the load of the initial state and the retry in case of error.
+ * @param props
+ */
+const useInitialValue = (props: Props) => {
+  const timerRetry = useRef<number | undefined>(undefined);
+  const navigation = useContext(NavigationContext);
+  const retry = () => {
+    timerRetry.current = undefined;
+    props.loadActualValue(props.hPan);
+  };
+
+  /**
+   * When the focus change, clear the timer (if any) and reset the value to undefined
+   * focus: true  -> a new schedule is allowed
+   * focus: false -> clear all the pending schedule
+   */
+  useEffect(() => {
+    clearTimeout(timerRetry.current);
+    timerRetry.current = undefined;
+  }, [navigation.isFocused()]);
+
+  useEffect(() => {
+    // Initial state, request the state
+    if (props.bpdPotActivation === pot.none) {
+      props.loadActualValue(props.hPan);
+    } else if (
+      pot.isNone(props.bpdPotActivation) &&
+      pot.isError(props.bpdPotActivation) &&
+      timerRetry.current === undefined &&
+      navigation.isFocused()
+    ) {
+      // If the pot is NoneError, the navigation focus is on the element
+      // and no other retry are scheduled
+      timerRetry.current = setTimeout(retry, retryTimeout);
+    }
+  }, [props.bpdPotActivation, timerRetry.current, navigation.isFocused()]);
+
+  // Component unmount, clear scheduled
+  useEffect(
+    () => () => {
```

Comment:
```suggestion
    () => {
```

Reply:
This is intended, because I need to return a function that should be called when the component is unmounted.